File: frontend/orcamentos/lib/app/data/model/fin_natureza_financeira_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:orcamentos/app/data/domain/domain_imports.dart';

class FinNaturezaFinanceiraModel {
	int? id;
	String? codigo;
	String? descricao;
	String? tipo;
	String? aplicacao;

	FinNaturezaFinanceiraModel({
		this.id,
		this.codigo,
		this.descricao,
		this.tipo,
		this.aplicacao,
	});

	static List<String> dbColumns = <String>[
		'id',
		'codigo',
		'descricao',
		'tipo',
		'aplicacao',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Codigo',
		'Descricao',
		'Tipo',
		'Aplicacao',
	];

	FinNaturezaFinanceiraModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		codigo = FinNaturezaFinanceiraDomain.getCodigo(jsonData['codigo']);
		descricao = jsonData['descricao'];
		tipo = FinNaturezaFinanceiraDomain.getTipo(jsonData['tipo']);
		aplicacao = jsonData['aplicacao'];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['codigo'] = FinNaturezaFinanceiraDomain.setCodigo(codigo);
		jsonData['descricao'] = descricao;
		jsonData['tipo'] = FinNaturezaFinanceiraDomain.setTipo(tipo);
		jsonData['aplicacao'] = aplicacao;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		codigo = plutoRow.cells['codigo']?.value != '' ? plutoRow.cells['codigo']?.value : 'AAA';
		descricao = plutoRow.cells['descricao']?.value;
		tipo = plutoRow.cells['tipo']?.value != '' ? plutoRow.cells['tipo']?.value : 'AAA';
		aplicacao = plutoRow.cells['aplicacao']?.value;
	}	

	FinNaturezaFinanceiraModel clone() {
		return FinNaturezaFinanceiraModel(
			id: id,
			codigo: codigo,
			descricao: descricao,
			tipo: tipo,
			aplicacao: aplicacao,
		);			
	}

	
}

File: backend/financeiro/src/model/fin_natureza_financeira_model.py
from src import db


class FinNaturezaFinanceiraModel(db.Model):
    __tablename__ = 'fin_natureza_financeira'

    id = db.Column(db.Integer, primary_key=True)
    codigo = db.Column(db.String(4))
    tipo = db.Column(db.String(1))
    descricao = db.Column(db.String(100))
    aplicacao = db.Column(db.String(250))


    def mapping(self, data):
        self.id = data.get('id')
        self.codigo = data.get('codigo')
        self.tipo = data.get('tipo')
        self.descricao = data.get('descricao')
        self.aplicacao = data.get('aplicacao')

    def serialize(self):
        return {
            'id': self.id,
            'codigo': self.codigo,
            'tipo': self.tipo,
            'descricao': self.descricao,
            'aplicacao': self.aplicacao,
        }